src/config.ts

```diff
@@ -20,7 +20,8 @@ const DefaultConfig: Partial<Libp2pInit> = {
     announceFilter: (multiaddrs: Multiaddr[]) => multiaddrs
   },
   connectionManager: {
-    maxConnections: 300,
+    maxIncomingConnections: 300,
+    maxOutgoingConnections: 300,
```

Comment:
This effectively doubles the number of allowed connections, I'm not sure if that's a good idea.

Reply:
There are valid use cases for expecting mostly incoming connections, it may be a better approach to just always ensure we can make outgoing connections to prevent eclipse attacks?

Reply:
If a consumer has a set of trusted peers and IP addresses, we can actually always have outgoing connections to these trusted peers by adding them to the `AllowList` (which https://github.com/libp2p/js-libp2p/issues/1515 consolidates by ensuring they are never pruned), so that should mitigate against eclipse attacks. I know that [go-lip2p's resource manager is configured with limits on the number of outgoing and incoming connections](https://github.com/libp2p/go-libp2p/blob/97607949208fc95bea759eeaa88db3ad2c86bf38/p2p/host/resource-manager/limit.go#L89)